pkg/reconciler/ingress/ingress.go

```diff
@@ -0,0 +1,151 @@
+type Reconciler struct {
+	statusManager status.Manager
+
+	gwapiclient gwapiclientset.Interface
+
+	// Listers index properties about resources
+	httprouteLister gwlisters.HTTPRouteLister
+}
+
+var (
+	_ ingressreconciler.Interface = (*Reconciler)(nil)
+)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (c *Reconciler) ReconcileKind(ctx context.Context, ingress *v1alpha1.Ingress) pkgreconciler.Event {
+	reconcileErr := c.reconcileIngress(ctx, ingress)
+	if reconcileErr != nil {
+		ingress.Status.MarkIngressNotReady(notReconciledReason, notReconciledMessage)
+		return reconcileErr
+	}
+
+	return nil
+}
+
+func (c *Reconciler) reconcileIngress(ctx context.Context, ing *v1alpha1.Ingress) error {
+	logger := logging.FromContext(ctx)
+
+	// We may be reading a version of the object that was stored at an older version
+	// and may not have had all of the assumed defaults specified.  This won't result
+	// in this getting written back to the API Server, but lets downstream logic make
+	// assumptions about defaulting.
+	ing.SetDefaults(ctx)
+	before := ing.DeepCopy()
+
+	ing.Status.InitializeConditions()
+
+	if _, err := ingress.InsertProbe(ing); err != nil {
+		return fmt.Errorf("failed to add knative probe header: %w", err)
+	}
+
+	logger.Infof("Reconciling ingress: %#v", ing)
+
+	for _, rule := range ing.Spec.Rules {
+		rule := rule
+
+		httproutes, err := c.reconcileHTTPRoute(ctx, ing, &rule)
+		if err != nil {
+			return err
+		}
+
+		if isHTTPRouteReady(httproutes) {
+			ing.Status.MarkNetworkConfigured()
+		} else {
+			ing.Status.MarkIngressNotReady("HTTPRouteNotReady", "Waiting for HTTPRoute becomes Ready.")
+		}
+		logger.Infof("HTTPRoute successfully synced %v", httproutes)
+	}
+
+	ready, err := c.statusManager.IsReady(ctx, before)
+	if err != nil {
+		return fmt.Errorf("failed to probe Ingress: %w", err)
+	}
+
+	if ready {
+		gatewayConfig := config.FromContext(ctx).Gateway
+
+		namespacedNameService := gatewayConfig.Gateways[v1alpha1.IngressVisibilityExternalIP].Service
+		publicLbs := []v1alpha1.LoadBalancerIngressStatus{
+			{DomainInternal: network.GetServiceHostname(namespacedNameService.Name, namespacedNameService.Namespace)},
+		}
+
+		namespacedNameLocalService := gatewayConfig.Gateways[v1alpha1.IngressVisibilityClusterLocal].Service
+		privateLbs := []v1alpha1.LoadBalancerIngressStatus{
+			{DomainInternal: network.GetServiceHostname(namespacedNameLocalService.Name, namespacedNameLocalService.Namespace)},
+		}
+
+		ing.Status.MarkLoadBalancerReady(publicLbs, privateLbs)
+	} else {
+		ing.Status.MarkLoadBalancerNotReady()
+	}
+
+	return nil
+}
+
+// isHTTPRouteReady will check the status conditions of the ingress and return true if
+// all gateways have been admitted.
+func isHTTPRouteReady(r *gatewayv1alpha1.HTTPRoute) bool {
+	if r.Status.Gateways == nil {
+		return false
+	}
+	for _, gw := range r.Status.Gateways {
+		if !isGatewayAdmitted(gw) {
+			// Return false if _any_ of the gateways isn't admitted yet.
```

Comment:
we may want to add a log for debugging purpose.

Reply:
KIngress's status will be updated by the https://github.com/nak3/net-ingressv2-1/blob/add-controller-code/pkg/reconciler/ingress/ingress.go#L98 when gateway is not admitted so probably the log is fine :thinking: